jetty-core/jetty-server/src/main/java/org/eclipse/jetty/server/handler/StatisticsHandler.java

```diff
@@ -54,17 +58,27 @@ public StatisticsHandler(Handler handler)
         super(handler);
     }
 
+    @Override
+    protected void doStart() throws Exception
+    {
+        reset();
+        super.doStart();
+    }
+
     @Override
     protected void onBeforeHandling(Request request)
     {
         _requestStats.increment();
+        _handlingStats.increment();
     }
 
     @Override
     protected void onAfterHandling(Request request, boolean handled, Throwable failure)
     {
         if (failure != null)
             _handlingFailures.increment();
+        _handlingStats.decrement();
+        _handlingTimeStats.record(NanoTime.since(request.getBeginNanoTime()));
```

Comment:
Recording a timestamp in `beforeHandle` would be mildly costly as it would need to be kept in a concurrent map. I've used `getHeadersNanoTime()` instead, which gets taken shortly before `handle()` is called, that should be accurate enough IMHO.